internal/github/publish_github_gist.go

```diff
@@ -0,0 +1,202 @@
+	if !ok {
+		return
+	}
+	documentName := metaMap.(plugin.MapData)["name"]
+	docName, ok := documentName.(plugin.StringData)
+	if !ok {
+		return
+	}
+	name = string(docName)
+	return
+}
+
+func publishGithubGist(loader ClientLoaderFn) plugin.PublishFunc {
+	// TODO: confirm if to be passed from the caller
+	logger := slog.New(slog.NewTextHandler(io.Discard, nil))
+	tracer := nooptrace.Tracer{}
+	return func(ctx context.Context, params *plugin.PublishParams) diagnostics.Diag {
+		document := parseContent(params.DataContext)
+		if document == nil {
+			return diagnostics.Diag{{
+				Severity: hcl.DiagError,
+				Summary:  "Failed to parse document",
+				Detail:   "document is required",
+			}}
+		}
+		datactx := params.DataContext
+		datactx["format"] = plugin.StringData(params.Format.String())
+		var printer print.Printer
+		switch params.Format {
+		case plugin.OutputFormatMD:
+			printer = mdprint.New()
+		case plugin.OutputFormatHTML:
+			printer = htmlprint.New()
+		default:
+			return diagnostics.Diag{{
+				Severity: hcl.DiagError,
+				Summary:  "Unsupported format",
+				Detail:   "Only md and html formats are supported",
+			}}
+		}
+		printer = print.WithLogging(printer, logger, slog.String("format", params.Format.String()))
+		printer = print.WithTracing(printer, tracer, attribute.String("format", params.Format.String()))
+
+		buff := bytes.NewBuffer(nil)
+		err := printer.Print(ctx, buff, document)
+		if err != nil {
+			return diagnostics.Diag{{
+				Severity: hcl.DiagError,
+				Summary:  "Failed to write to a file",
+				Detail:   err.Error(),
+			}}
+		}
+
+		client := loader(params.Config.GetAttr("github_token").AsString())
+
+		fileName := parseName(params.DataContext) + "." + params.Format.String()
+		filenameAttr := params.Args.GetAttr("filename")
+		if !filenameAttr.IsNull() && filenameAttr.AsString() != "" {
+			fileName = filenameAttr.AsString()
+		}
+		payload := &gh.Gist{
+			Public: gh.Bool(false),
+			Files: map[gh.GistFilename]gh.GistFile{
+				gh.GistFilename(fileName): {
+					Content:  gh.String(buff.String()),
+					Filename: gh.String(fileName),
+				},
+			},
+		}
+		// overrides params if defined
+		descriptionAttr := params.Args.GetAttr("description")
+		if !descriptionAttr.IsNull() && descriptionAttr.AsString() != "" {
+			payload.Description = gh.String(descriptionAttr.AsString())
+		}
+		makePublicAttr := params.Args.GetAttr("make_public")
+		if !makePublicAttr.IsNull() {
+			payload.Public = gh.Bool(makePublicAttr.True())
+		}
+		slog.InfoContext(ctx, "Publish to github gist", "filename", fileName)
+		gistId := params.Args.GetAttr("gist_id")
+		if gistId.IsNull() || gistId.AsString() == "" {
+			gist, _, err := client.Gists().Create(ctx, payload)
+			if err != nil {
+				return diagnostics.Diag{{
+					Severity: hcl.DiagError,
+					Summary:  "Failed to create gist",
+					Detail:   err.Error(),
+				}}
+			}
+			slog.InfoContext(ctx, "created gist", "url", *gist.HTMLURL)
```

Comment:
```suggestion
			slog.InfoContext(ctx, "Created gist", "url", *gist.HTMLURL)
```

Log messages can be full sentences, so it's better to capitalize the first letter (as opposed to Go errors)